File: 백준/Silver/1149. RGB거리/RGB거리.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[][] cost = new int[n + 1][3];
        int[][] dp = new int[n + 1][3];

        for (int i = 1; i <= n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            cost[i][0] = Integer.parseInt(st.nextToken());
            cost[i][1] = Integer.parseInt(st.nextToken());
            cost[i][2] = Integer.parseInt(st.nextToken());
        }

        dp[1][0] = cost[1][0];
        dp[1][1] = cost[1][1];
        dp[1][2] = cost[1][2];

        for (int i = 2; i <= n; i++) {
            dp[i][0] = Math.min(dp[i - 1][1], dp[i - 1][2]) + cost[i][0];
            dp[i][1] = Math.min(dp[i - 1][0], dp[i - 1][2]) + cost[i][1];
            dp[i][2] = Math.min(dp[i - 1][0], dp[i - 1][1]) + cost[i][2];
        }

        System.out.print(Math.min(dp[n][0], Math.min(dp[n][1], dp[n][2])));
    }
}

File: 백준/Silver/1149. RGB거리/RGB거리.kt
import java.util.*
import kotlin.math.min

fun main() {
    val n = readln().toInt()
    val arrRed = IntArray(1002)
    val arrBlue = IntArray(1002)
    val arrGreen = IntArray(1002)
    for (i in 1..n){
        val num = readln().split(" ").map { it.toInt() }
        arrRed[i] = num[0]
        arrBlue[i] = num[1]
        arrGreen[i] = num[2]
    }
    val dp = Array(1002) { IntArray(3) }
    dp[1][0] = arrRed[1]
    dp[1][1] = arrGreen[1]
    dp[1][2] = arrBlue[1]
    for (i in 2..n) {
        dp[i][0] = min(dp[i - 1][1], dp[i - 1][2]) + arrRed[i]
        dp[i][1] = min(dp[i - 1][0], dp[i - 1][2]) + arrGreen[i]
        dp[i][2] = min(dp[i - 1][0], dp[i - 1][1]) + arrBlue[i]
    }
    println(dp[n].minBy { it })
}